generation/v3.go

```diff
@@ -0,0 +1,953 @@
+	"⒊",
+	"⒋",
+	"⒌",
+	"⒍",
+	"⒎",
+	"⒏",
+	"⒐",
+	"⒑",
+	"⒒",
+	"⒓",
+	"⒔",
+	"⒕",
+	"⒖",
+	"⒗",
+	"⒘",
+	"⒙",
+	"⒚",
+	"⒛",
+	"⓪",
+	"⓫",
+	"⓬",
+	"⓭",
+	"⓮",
+	"⓯",
+	"⓰",
+	"⓱",
+	"⓲",
+	"⓳",
+	"⓴",
+	"❶",
+	"❷",
+	"❸",
+	"❹",
+	"❺",
+	"❻",
+	"❼",
+	"❽",
+	"❾",
+	"❿",
+	"֍",
+	"֎",
+	"ॐ",
+	"☤",
+	"☥",
+	"☦",
+	"☧",
+	"☨",
+	"☩",
+	"☪",
+	"☫",
+	"☬",
+	"☭",
+	"☮",
+	"☯",
+	"☸",
+	"✝",
+	"✞",
+	"✟",
+	"✠",
+	"✡",
+	"♩",
+	"♪",
+	"♫",
+	"♬",
+	"♭",
+	"♮",
+	"♯",
+	"🎹",
+	"🎼",
+	"࠽",
+	"℃",
+	"⌘",
+	"⌚",
+	"⌛",
+	"⌧",
+	"⌨",
+	"⌬",
+	"⍓",
+	"■",
+	"□",
+	"▯",
+	"◆",
+	"●",
+	"☀",
+	"☁",
+	"☂",
+	"☃",
+	"☄",
+	"☎",
+	"☏",
+	"☑",
+	"☒",
+	"☘",
+	"☠",
+	"☢",
+	"☣",
+	"☹",
+	"☺",
+	"☼",
+	"☽",
+	"☾",
+	"☿",
+	"♀",
+	"♁",
+	"♂",
+	"♨",
+	"♲",
+	"♻",
+	"♼",
+	"♽",
+	"♿",
+	"⚒",
+	"⚓",
+	"⚔",
+	"⚕",
+	"⚖",
+	"⚘",
+	"⚙",
+	"⚚",
+	"⚛",
+	"⚜",
+	"⚞",
+	"⚟",
+	"⚠",
+	"⚡",
+	"⚢",
+	"⚣",
+	"⚤",
+	"⚥",
+	"⚦",
+	"⚧",
+	"⚨",
+	"⚩",
+	"⚰",
+	"⚱",
+	"⚲",
+	"⚳",
+	"⚴",
+	"⚵",
+	"⛄",
+	"⛟",
+	"⛱",
+	"⛲",
+	"⛳",
+	"⛴",
+	"⛵",
+	"✀",
+	"✁",
+	"✂",
+	"✃",
+	"✄",
+	"✆",
+	"✈",
+	"✉",
+	"✊",
+	"✋",
+	"✌",
+	"✍",
+	"✎",
+	"✏",
+	"✐",
+	"✓",
+	"✔",
+	"✕",
+	"✖",
+	"✗",
+	"✘",
+	"✿",
+	"❀",
+	"❁",
+	"❄",
+	"❍",
+	"❏",
+	"❐",
+	"❑",
+	"❒",
+	"❖",
+	"⧫",
+	"⬧",
+	"🔒",
+	"₠",
+	"₡",
+	"₤",
+	"㍱",
+	"㍲",
+	"㍳",
+	"㍴",
+	"㍵",
+	"㍶",
+	"㎀",
+	"㎁",
+	"㎂",
+	"㎃",
+	"㎄",
+	"㎅",
+	"㎆",
+	"㎇",
+	"㎈",
+	"㎉",
+	"㎊",
+	"㎋",
+	"㎌",
+	"㎍",
+	"㎎",
+	"㎏",
+	"㎐",
+	"㎑",
+	"㎒",
+	"㎓",
+	"㎔",
+	"㎕",
+	"㎖",
+	"㎗",
+	"㎘",
+	"㎙",
+	"㎚",
+	"㎛",
+	"㎜",
+	"㎝",
+	"㎞",
+	"㎟",
+	"㎠",
+	"㎡",
+	"㎢",
+	"㎣",
+	"㎤",
+	"㎥",
+	"㎦",
+	"㎧",
+	"㎨",
+	"㎩",
+	"㎪",
+	"㎫",
+	"㎬",
+	"㎭",
+	"㎮",
+	"㎯",
+	"㎰",
+	"㎱",
+	"㎲",
+	"㎳",
+	"㎴",
+	"㎵",
+	"㎶",
+	"㎷",
+	"㎸",
+	"㎹",
+	"㎺",
+	"㎻",
+	"㎼",
+	"㎽",
+	"㎾",
+	"㎿",
+	"㏀",
+	"㏁",
+	"㏂",
+	"㏃",
+	"㏄",
+	"㏅",
+	"㏆",
+	"㏇",
+	"㏈",
+	"㏉",
+	"㏊",
+	"㏋",
+	"㏌",
+	"㏍",
+	"㏎",
+	"㏏",
+	"㏐",
+	"㏑",
+	"㏒",
+	"㏓",
+	"㏔",
+	"㏕",
+	"㏖",
+	"㏗",
+	"㏘",
+	"㏙",
+	"㏚",
+	"㏛",
+	"㏜",
+	"㏝",
+}
+
+// V3 convert [r,g,b,index] tuple into [rgb_r,rgb_g,rgb_b,hsl_h,hsl_s,hsl_l,lab_l,lab_a,lab_b] tuple
+// Index is used to generate a different symbol based on the csv.
+func V3(rgbR, rgbG, rgbB, index int) ([]string, error) {
+	symbol, err := getCurrentSymbol(index)
+	if err != nil {
+		return nil, err
+	}
+
+	hslH, hslS, hslL, err := hsl(rgbR, rgbG, rgbB)
+	if err != nil {
+		return nil, err
+	}
+	labL, labA, labB, err := lab(rgbR, rgbG, rgbB)
+	if err != nil {
+		return nil, err
+	}
+	return []string{
+		fmt.Sprintf("%s", symbol),
+		fmt.Sprintf("%d", rgbR),
+		fmt.Sprintf("%d", rgbG),
+		fmt.Sprintf("%d", rgbB),
+		fmt.Sprintf("%.2f", hslH),
+		fmt.Sprintf("%.2f", hslS),
+		fmt.Sprintf("%.2f", hslL),
+		fmt.Sprintf("%.2f", labL),
+		fmt.Sprintf("%.2f", labA),
+		fmt.Sprintf("%.2f", labB),
+	}, nil
+}
+
+// Gets symbol for this row based on the index provided
+func getCurrentSymbol(index int) (string, error) {
+	if len(symbols) < index {
+		return "", errors.New("No symbol with index found")
```

Comment:
Being picky here, but as state in the [documentation](https://github.com/golang/go/wiki/Errors)

> Error strings should not start with a capital letter because they'll often be prefixed before printing

:wink:

Reply:
No problem. Good thing to know for future. I've updated this now.